repotracker/repotracker.go

```diff
@@ -889,21 +900,43 @@ func createVersionItems(ctx context.Context, v *model.Version, ref *model.Projec
 		return false, nil
 	}
 
-	return client.UseSession(ctx, func(sessCtx mongo.SessionContext) error {
-		for i := 0; i < retryCount; i++ {
+	return transactionWithRetries(ctx, txFunc)
+}
+
+// If we error in aborting transaction, we recreate the client and start again.
+// If we abort successfully and the error is a transient transaction error, we retry using the same client.
+func transactionWithRetries(ctx context.Context, txFunc func(sessCtx mongo.SessionContext) (bool, error)) error {
+	const transactionRetryCount = 5
+	const clientRetryCount = 2
```

Comment:
I don't think separating these is needed - just retry 5 times with a new session each time

Reply:
This feels lazy to me--why bother aborting transaction and looking at transaction errors then?

Reply:
My point was to retry with a new session if it's a transient error, so it still requires looking at the error. I'm just thinking it's not worth trying to figure out when we can retry the transaction with the same session, unless the overhead of creating a new session is large